students/ThanhNhan/lesson06/test_mailroom.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python
+import mailroom4 as mr4
+import os
+
+"""
+Send a thank you (adds a new donor or updates existing donor info)
+Create a report
+Send letters (creates files)
+Your unit tests should test the data manipulation logic code: generating thank you text, adding or updating donors, and listing donors.
+Send a thank you (adds a new donor or updates existing donor info)
+"""
+def test_existing_donor():
+
+    new_donor = {"Tom Cruise": [233.23, 12.17],
+             "Alyssa Nhan": [23., 23],
+             "Jessy Test": [33.0],
+             "Jin Unknown": [12.0, 12.0]
+             }
+    mr4.donor_dict = new_donor
+    new_donor_db = mr4.chk_exist_donor()
+    assert new_donor_db == ["Tom Cruise", "Alyssa Nhan", "Jessy Test", "Jin Unknown"]
+
+
+# Validate Name Input
+def test_validate_name():
+    name1 = mr4.validate_name(" ")
+    assert name1 is False
+
+    name2 = mr4.validate_name("23")
+    assert name2 is False
+
+    name3 = mr4.validate_name("asd2a adf3")
+    assert name3 is False
+
+    name4 = mr4.validate_name("3ass3 12sdf")
+    assert name4 is False
+
+    name5 = mr4.validate_name("Tom Asd")
+    assert name5 is True
+
+    name6 = mr4.validate_name("Tom")
+    assert name6 is False
+
+    name7 = mr4.validate_name("Tom ")
+    assert name7 is False
+
+    name8 = mr4.validate_name("Tom Asd III")
+    assert name8 is True
+
+    name9 = mr4.validate_name("Tom Asdasfasf, III")
+    assert name9 is True
+
+# Test for lookup Name
+def test_lookup_name():
+    name1 = mr4.lookup_name("Alyssa Nhan")
+    assert name1 == "Alyssa Nhan"
+
+    name2 = mr4.lookup_name("Thanh Inte")
+    assert name2 is None
+
+#Test dollar amount input
+def test_ck_char_input():
+    donate1 = mr4.ck_dollar("xasdf")
+    assert donate1 is False
+def test_ck_integer():
+    donate2 = mr4.ck_dollar("500")
+    assert donate2 is True
+def test_ck_longfloat():
+    donate3 = mr4.ck_dollar("0.001")
+    assert donate3 is False
+def test_ck_string():
+    donate4 = mr4.ck_dollar("asd")
+    assert donate4 is False
+def test_ck_pt2float():
+    donate4 = mr4.ck_dollar("0.23")
+    assert donate4 is True
+
+#Test for donation add
+def test_existing_donation():
+    mr4.add_donation("Alyssa Nhan", "Alyssa Nhan", 2.32)
+    assert mr4.donor_dict["Alyssa Nhan"] == [23., 23, 2.32]
```

Comment:
careful here -- this is assuming a particular state before this test runs. If another test gets added later that adds a donation to this donor, this test will fail, even though it is working correctly.